### PR TITLE
Gitops 1323 config dynamic plugin

### PR DESCRIPTION
**Pull request by ciiay**
**What type of PR is this?**
/kind enhancement

**What does this PR do / why we need it**:
GITOPS-1323, pull gitops-plugin image into operator.

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #? [GITOPS-1323](https://issues.redhat.com/browse/GITOPS-1323)

**Test acceptance criteria**:

* [x] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:
1. Create a docker image base off this branch code and install it to a cluster
2. Install pipelines-operator
3. Apply argocd manifests
4. Observe the Dynamic Plugin is available on Developer/Environment - Dynamic

### CODE REVIEW

File: bundle/manifests/pipelines.openshift.io_gitopsservices.yaml [hunk 1]
```diff
@@ -34,13 +34,19 @@ spec:
           spec:
             description: GitopsServiceSpec defines the desired state of GitopsService
             properties:
```
Comment (jopit) [inline bundle/manifests/pipelines.openshift.io_gitopsservices.yaml:36]: Not sure why this NodeSelector code is part of this PR?

File: controllers/gitopsservice_controller.go [hunk 2]
```diff
@@ -344,6 +349,15 @@ func (r *ReconcileGitopsService) reconcileDefaultArgoCDInstance(instance *pipeli
 			defaultArgoCDInstance.Spec.NodePlacement.NodeSelector = argocdutil.AppendStringMap(defaultArgoCDInstance.Spec.NodePlacement.NodeSelector, instance.Spec.NodeSelector)
 		}
 	}
+	if len(instance.Spec.NodeSelector) > 0 {
```
Comment (jopit) [inline controllers/gitopsservice_controller.go:352]: Again, should this NodeSelector related code be part of this PR?

File: bundle/manifests/pipelines.openshift.io_gitopsservices.yaml (hunk 1 above)
Comment (ciiay) [reply bundle/manifests/pipelines.openshift.io_gitopsservices.yaml:36]: Thanks for the review. I got some code wrong when I rebased my branch. It should be fixed now.

File: controllers/gitopsservice_controller.go (hunk 2 above)
Comment (ciiay) [reply controllers/gitopsservice_controller.go]: Thanks for the catch. This should be fixed now.

**Review by jopit — CHANGES_REQUESTED** [review]

File: controllers/consoleplugin.go [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+	kubeAppLabelInstance         = "app.kubernetes.io/instance"
+	kubeAppLabelPartOf           = "app.kubernetes.io/part-of"
+	kubeAppLabelRuntimeNamespace = "app.kubernetes.io/runtime-namespace"
+	httpdConfigMapName           = "httpd-cfg"
+)
+
+func pluginDeployment() *appsv1.Deployment {
+	consolePluginImage := os.Getenv(pluginImageEnv)
+	if consolePluginImage == "" {
+		consolePluginImage = "quay.io/redhat-developer/gitops-console-plugin"
```
Comment (iam-veeramalla) [inline controllers/consoleplugin.go:46]: This should be a variable or a constant and split into two parts

Image := quay.io/redhat-developer/gitops-console-plugin 
version := "<whatever is the version>"

File: controllers/consoleplugin.go [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+					},
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("128Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("250m"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("256Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("500m"),
```
Comment (iam-veeramalla) [inline controllers/consoleplugin.go:88]: How did we derive the resource requests and limits values ? Any research ?

File: controllers/consoleplugin.go [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+	kubeAppLabelRuntimeNamespace = "app.kubernetes.io/runtime-namespace"
+	httpdConfigMapName           = "httpd-cfg"
+)
+
+func pluginDeployment() *appsv1.Deployment {
+	consolePluginImage := os.Getenv(pluginImageEnv)
+	if consolePluginImage == "" {
+		consolePluginImage = "quay.io/redhat-developer/gitops-console-plugin"
+	}
+	podSpec := corev1.PodSpec{
```
Comment (iam-veeramalla) [inline controllers/consoleplugin.go:48]: Can you please test this pod specification/deployment on a 4.12 cluster which is based on k8s v1.25 to verify if we need to include podSecurity ?

https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

File: controllers/consoleplugin.go [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+	template := corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				kubeAppLabelApp: gitopsPluginName,
+			},
+		},
+		Spec: podSpec,
+	}
+	var replicas int32 = 1
+	return &appsv1.Deployment{
```
Comment (iam-veeramalla) [inline controllers/consoleplugin.go:128]: Can this be split into different functions ?

podSpec := getPluginPodSpec()
getPluginDeployment(podSpec)

**Review by iam-veeramalla — CHANGES_REQUESTED** [review]

File: controllers/consoleplugin.go (hunk 4 above)
Comment (ciiay) [reply controllers/consoleplugin.go:88]: I copied these settings from kam.go. How should I decide the value?

Comment (iam-veeramalla) [reply controllers/consoleplugin.go:88]: You can create a  resource quota of 20 CPU and 20 GB on openshift-gitops namespace
after that, all the pods in the namespace can be created only if thy have resource requests and limits in their specification, just try to keep the resource limits as low as posible and run trail and error

this just for your understanding.

As of this PR, the requests and limits seems to be low already. I am ok with them

**Comment by jaideepr97** [comment]
@ciiay you will have to have all your commits signed off on to pass the DCO check

File: controllers/consoleplugin.go (hunk 3 above)
Comment (ciiay) [reply controllers/consoleplugin.go:46]: @jaideepr97 In what case we need to put default value in `common.go` ? Or just all default value should be in `common.go`?

Comment (jaideepr97) [reply controllers/consoleplugin.go:46]: Yeah, I think all hard coded default values should be tracked in common.go or defaults.go

That way it is easier to make changes in the future and we don't have to hunt down where individual strings are stored

**Review by iam-veeramalla — APPROVED** [review]
/lgtm
/approve

**Pull request merged**